### DISCUSSION TITLE
Is there a way to configure path to ledger binary?

### DISCUSSION THREAD

**Discussion by bhankas** [ananthakumaran/paisa#291] [Q&A]
I setup paisa service on my server a while ago, but just started using it. I see that a ledger binary is included and stored/executed from `~/.cache/paisa/ledger`. The problem is my OS of choice is NixOS and prebuilt binary does not work because the library paths in NixOS are nonstandard.

The simplest and dumbest solution that I find currently working is simple `rm -rf ~/.cache/paisa/ledger && ln -s /etc/profiles/per-user/bhankas/bin/ledger ~/.cache/paisa/ledger`.

However, this must be invoked after first exec ledger by paisa service. I added above command to `ExecStartPost=` of my systemd unit for paisa, it works, but on first exec the application replaces the binary again. 

Is there a way for me to configure ledger binary path permanently? For what its worth, `which ledger` returns the correct path, but this being NixOS its not from the [usual list that paisa looks for](https://github.com/ananthakumaran/paisa/blob/8430ad8ec6b7dabbf95fab1c1dae258f8dd319a9/docs/faq.md).

**Comment by ananthakumaran** [comment]
Install ledger on your system via nix, then paisa will use that instead of its own embedded ledger.

**Reply by bhankas** [reply]
Hi, thanks for prompt response and apologies for taking long to respond.

This does not work, unfortunately. I have to run `rm -rf ~/.cache/paisa/ledger && ln -s /run/current-system/sw/bin/ledger ~/.cache/paisa/ledger` otherwise I get `exec format error`.

ledger binary is installed system-wide by Nix and available to all users. Systemd unit is configured at user level like this:

```
[Install]
WantedBy=default.target

[Service]
ExecStart=/nix/store/ay6dh9djailikslfjkk7c5qbpg3yr45v-paisa-cli-0.7.0/bin/paisa serve
ExecStartPost=/nix/store/ssqdazslv2wjpybnk0pjwnpq0dfpa6q2-coreutils-9.5/bin/rm -rf /home/bhankas/.cache/paisa/ledger
ExecStartPost=/nix/store/ssqdazslv2wjpybnk0pjwnpq0dfpa6q2-coreutils-9.5/bin/ln -s /run/current-system/sw/bin/ledger /home/bhankas/.cache/paisa/ledger
Restart=always
WorkingDirectory=/home/bhankas/org/ledger

[Unit]
Description=Paisa – Personal Finance Manager.
```

The `ExecStartPost` were added to avoid manually removing/readding the symlink, but don't seem to have any effect.

IMO, the list of paths to look for approach works only so far. NixOS in generat is pretty weird about this. The above path (` /run/current-system/sw/bin/ledger`) itself is a symlink that points somewhere in Nix store (`/nix/store/{hash}-ledger/bin/ledger`), which is expected to change as often as system is updated.

It would be far more amenable to just have a global config to set this path to whatever the system provides. Maintaining the list of common paths then becomes almost unnecessary, but I'm open for inputs.
